src/Azure/Orleans.Transactions.AzureStorage/TransactionalState/AzureTableTransactionalStateStorage.cs

```diff
@@ -0,0 +1,184 @@
+                this.states = this.states.Where(p => string.Compare(p.TransactionId, transactionIdToCommit) >= 0).ToList();
+                Cleanup(dead).Ignore();
+                return this.key.ETag;
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogError("Transactional state confirm failed {Exception}.", ex);
+                throw;
+            }
+        }
+
+        private async Task<KeyEntity> ReadKey()
+        {
+            var query = new TableQuery<KeyEntity>()
+                .Where(AzureStorageUtils.PointQuery(this.partition, KeyEntity.RK));
+            TableQuerySegment<KeyEntity> queryResult = await table.ExecuteQuerySegmentedAsync(query, null).ConfigureAwait(false);
+            return queryResult.Results.Count == 0
+                ? new KeyEntity() { PartitionKey = this.partition }
+                : queryResult.Results[0];
+        }
+
+        private Task WriteKey()
+        {
+            return (string.IsNullOrEmpty(this.key.ETag))
+                ? this.table.ExecuteAsync(TableOperation.Insert(this.key))
+                : this.table.ExecuteAsync(TableOperation.Replace(this.key));
+        }
+
+        private async Task<List<StateEntity>> ReadStates()
+        {
+            var query = new TableQuery<StateEntity>()
+                .Where(AzureStorageUtils.RangeQuery(this.partition, StateEntity.RKMin, StateEntity.RKMax));
+            TableContinuationToken continuationToken = null;
+            List<StateEntity> results = new List<StateEntity>();
+            do
+            {
+                TableQuerySegment<StateEntity> queryResult = await table.ExecuteQuerySegmentedAsync(query, continuationToken).ConfigureAwait(false);
+                results.AddRange(queryResult.Results);
+                continuationToken = queryResult.ContinuationToken;
+            } while (continuationToken != null);
+            return results;
+        }
+
+        private bool TryFindState(string transactionId, out TState state)
+        {
+            state = null;
+            if (string.IsNullOrEmpty(transactionId))
+                return false;
+
+            StateEntity entity = this.states.FirstOrDefault(s => s.TransactionId == transactionId);
+            if (entity == null)
+            {
+                throw new InvalidOperationException($"Transactional state non-recoverable error.  Commited state for transaction {transactionId} not found.");
+            }
+            state = entity.GetState<TState>(this.JsonSettings);
+            return true;
+        }
+
+        private async Task Cleanup(List<StateEntity> deadStates)
+        {
+            try
+            {
+                var batchOperation = new TableBatchOperation();
+                deadStates.ForEach(batchOperation.Delete);
+                await table.ExecuteBatchAsync(batchOperation).ConfigureAwait(false);
```

Comment:
batch op size limit check here  ?

Reply:
Probably should break into multiple batches if above batch size limit.